--- partimage-0.6.9/src/client/Makefile.am.slang_static~	2010-07-25 17:30:31.000000000 +0200
+++ partimage-0.6.9/src/client/Makefile.am	2012-09-18 13:04:02.064830896 +0200
@@ -33,7 +33,7 @@ partimage_SOURCES =         \
     mbr_backup.cpp
 
 # all libs are added in LIBS, we needn't add them to partimage_LDADD 
-partimage_LDADD = -lslang fs/libfs.a ../shared/libshared.a
+partimage_LDADD = $(libdir)/libslang.a fs/libfs.a ../shared/libshared.a
 partimage_LDFLAGS = @PARTIMAGE_LDFLAGS@
 
 DEFS=@DEFS@ -DLOCALEDIR=\"${localedir}\" -D_REENTRANT -D_FILE_OFFSET_BITS=64
