diff -Nru partimage-0.6.9.orig/src/client/netclient.cpp partimage-0.6.9/src/client/netclient.cpp
--- partimage-0.6.9.orig/src/client/netclient.cpp	2010-07-25 17:30:31.000000000 +0200
+++ partimage-0.6.9/src/client/netclient.cpp	2016-03-03 05:49:43.134033802 +0100
@@ -43,7 +43,7 @@
     {
       showDebug(3, "initializing client ssl\n");
       SSLeay_add_ssl_algorithms();
-      meth = SSLv2_client_method();
+      meth = SSLv23_client_method();
       SSL_load_error_strings();
       ctx = SSL_CTX_new(meth);
       if (!ctx)
 
